components/user_groups_modal/user_groups_list/user_groups_list.tsx

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {Group, GroupPermissions} from 'mattermost-redux/types/groups';
+
+import NoResultsIndicator from 'components/no_results_indicator';
+import {NoResultsVariant} from 'components/no_results_indicator/types';
+import LoadingScreen from 'components/loading_screen';
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+
+import * as Utils from 'utils/utils.jsx';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import {ModalData} from 'types/actions';
+import {ModalIdentifiers} from 'utils/constants';
+import ViewUserGroupModal from 'components/view_user_group_modal';
+
+export type Props = {
+    groups: Group[];
+    searchTerm: string;
+    loading: boolean;
+    groupPermissionsMap: Record<string, GroupPermissions>;
+    onScroll: () => void;
+    onExited: () => void;
+    backButtonAction: () => void;
+    actions: {
+        archiveGroup: (groupId: string) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+const UserGroupsList = React.forwardRef((props: Props, ref?: React.Ref<HTMLDivElement>) => {
+    const {
+        groups,
+        searchTerm,
+        loading,
+        groupPermissionsMap,
+        onScroll,
+        backButtonAction,
+        onExited,
+        actions,
+    } = props;
+
+    const archiveGroup = async (groupId: string) => {
+        await actions.archiveGroup(groupId);
+    };
+
+    const goToViewGroupModal = (group: Group) => {
+        actions.openModal({
+            modalId: ModalIdentifiers.VIEW_USER_GROUP,
+            dialogType: ViewUserGroupModal,
+            dialogProps: {
+                groupId: group.id,
+                backButtonCallback: backButtonAction,
+                backButtonAction: () => {
+                    goToViewGroupModal(group);
+                },
+            },
+        });
+        onExited();
+    };
+
+    return (
+        <div
+            className='user-groups-modal__content user-groups-list'
+            onScroll={onScroll}
+            ref={ref}
+        >
+            {(groups.length === 0 && searchTerm) &&
+                <NoResultsIndicator
+                    variant={NoResultsVariant.ChannelSearch}
+                    titleValues={{channelName: `"${searchTerm}"`}}
+                />
+            }
+            {groups.map((group) => {
```

Comment:
nit: Probably you can change this slightly to something like:

```js
{groups.map((group) => (
    <div>
        ....
    </div>
)}
```

It is equivalent, but remove some indentation and looks a bit nicer.

You can also think about the possibility of moving that code in another functional component, because looks like a ton of JSX that can be extracted easily.